doc/developer/design/20230531_system_privileges.md

```diff
@@ -0,0 +1,86 @@
+# System Privileges
+
+- Associated: [RBAC](20230216_role_based_access_control.md)
+
+## Context
+
+Role based access control comprises a set of features that allow customers to control the
+privileges of individual users. Object privileges control user's access to database objects, they
+are granted with the `GRANT` keyword and revoked with the `REVOKE` keyword. For example,
+`GRANT SELECT ON TABLE t TO r` will grant the ability to read table `t` to role `r`. `GRANT CREATE
+ON SCHEMA s TO r` will grant the ability to create new objects in schema `s` to role `r`. Object
+privileges can be inherited through role membership. Role attributes control the user's ability to
+create (and sometimes modify) top level objects, they are granted and revoked with the `ALTER ROLE`
+keywords. For example, `ALTER ROLE r CREATEDB` will grant the ability to create databases to
+role `r`. `ALTER ROLE r CREATECLUSTER` will grant the ability to create clusters to role `r`. Role
+attributes cannot be inherited through role membership.
+
+Role attributes present some problems as they currently exist:
+
+- It can be confusing that there are two separate privilege primitives that control a role's
+  privilege.
+- Role attributes are not inherited which makes it cumbersome to grant attributes to a large
```

Comment:
Can you add a point here and then in goals that describes what a user wants to accomplish that is hard, and then a goal that meets that user need? I assume it'd just be this sentence reframed, but I want something coming from a user's perspective.

Reply:
Done, let me know what you think.

Reply:
I don't see this addition anywhere. Am I missing it?

Reply:
New changes look great.